back/engines/commercial/smart_groups/app/lib/smart_groups/rules/email.rb

```diff
@@ -50,7 +50,18 @@ def initialize(predicate, value)
       self.value = value
     end
 
+    def cache_key_fragments
+      [User.all.cache_key_with_version, predicate, value]
+    end
+
     def filter(users_scope)
+      member_ids = Rails.cache.fetch(cache_key) do
```

Comment:
It does, our cache store is just limited in size FIFO style. When memcached hits its memory limit, it will automatically expire the oldest / least used entries first.

Reply:
Ah, I see. I'm just curious, where is it deployed? Can we see some stats (e.g., server load/bandwidth, the oldest key age)? There seems to be no info in Notion.

Reply:
It's part of [the stack](https://github.com/CitizenLabDotCo/cl2-deployment/blob/c68d90b7ca69ba1258030fd81ccd823f6e79c7ab/docker-compose-production.yml#L93)

No statistics at the moment.